libs/api/domains/national-registry-x-road/src/lib/nationalRegistryXRoad.service.ts

```diff
@@ -46,6 +51,47 @@ export class NationalRegistryXRoadService {
     }
   }
 
+  async getNationalRegistryResidenceHistory(
+    nationalId: string,
+    authToken: string,
+  ): Promise<NationalRegistryResidenceHistory> {
+    const historyList = await this.nationalRegistryFetch<Array<Heimili>>(
+      `/${nationalId}/buseta`,
+      authToken,
+    )
+
+    const history = historyList.map((heimili: Heimili) => {
+      // API says Date, but is string -- fallback in case that changes in the future
+      const date =
+        typeof heimili.breytt === 'string'
+          ? new Date(heimili.breytt)
+          : heimili.breytt
+      return {
+        address: {
+          city: heimili.stadur,
+          postalCode: heimili.postnumer,
+          streetName: heimili.heimilisfang,
+        } as NationalRegistryAddress,
+        country: heimili.landakodi,
+        dateOfChange: date,
+      } as NationalRegistryResidence
+    })
+
+    const daysPerCountry = computeCountryResidence(history)
+    const countryResidenceDays = daysPerCountry
+      ? Object.entries(daysPerCountry).map(([countryCode, days]) => ({
+          countryCode,
+          days,
+        }))
+      : null
```

Comment:
What is happening here?

Reply:
the calculation functions returns a Record<string, number> and I don't really know how to express that as a type in graphql, so i'm converting it to more normal `NationalRegistryCountryResidenceDays` type.

this function converts it between those two formats

Reply:
I understand, then the `computeCountryResidence` function should return that type. Instead of that returning another type that always need to be transformed.